### PR TITLE
CASSANDRA-20876: Offline nodetool commands should not print network options in help

### PR DESCRIPTION
**Pull request by arvindKandpal-ksolves**
## Summary
Offline nodetool commands (e.g. `history`) do not require a JMX connection, 
yet their `--help` output was showing network-related options like `--host`, 
`--port`, `--password`, etc. This is misleading and unnecessary.

## Changes
- **`AbstractCommand.java`**: Added a public `isOfflineCommand()` method that 
  safely wraps the existing `protected shouldConnect()` to expose whether a 
  command requires a JMX connection.
- **`CassandraCliHelpLayout.java`**: Updated `parentCommandOptionsWithJmxOptions()` 
  to suppress `JmxConnect` options from help output when the command is offline. 
  `NodetoolCommand` options are still always shown for backwards compatibility.
- **`test/resources/nodetool/help/history`**: Updated help snapshot to reflect 
  the corrected output — JMX network options are no longer shown for `history`.

patch by Arvind Kandpal; reviewed by TBD for CASSANDRA-20876

### CODE REVIEW

File: src/java/org/apache/cassandra/tools/nodetool/AbstractCommand.java [hunk 1]
```diff
@@ -78,6 +78,23 @@ protected boolean shouldConnect() throws ExecutionException
...
 
+    /**
+     * Check if the command is designed to run locally without a JMX connection.
+     * @return {@code true} if it is an offline command, {@code false} otherwise.
+     */
+    public boolean isOfflineCommand()
+    {
+        try
+        {
+            return !shouldConnect();
```
Comment (Mmuzaf) [inline src/java/org/apache/cassandra/tools/nodetool/AbstractCommand.java]: @arvindKandpal-ksolves The problem with the current approach is exactly that `isOfflineCommand()` calls `shouldConnect()`, which is an execution-flow method, intentionally runtim-overridable and side-effecting (that's why it declares throws ExecutionException). See the `Sjk` (where the 'locality' is runtime dependent) command and check the test `NodetoolHelpCommandsOutputTest`.

As for the `history` command we know in advance that it is local, and this knowledge should be a part of command metdata. 

So the right fix is to make this knowledge to be a part of command hierary: so either create a `LocalCommand` marker interface, or `AbstractLocalCommand extends AbstractCommand`. Then in the Layout class, check whether the command is local by examining the class it extends.

**Comment by arvindKandpal-ksolves** [comment]
Thanks for the feedback @Mmuzaf! 

You're completely right about the potential side effects of calling `shouldConnect()`. I've updated the PR based on your suggestion: I reverted the previous approach, introduced a `LocalCommand` marker interface, made the `History` command implement it, and updated `CassandraCliHelpLayout` to just check `instanceof LocalCommand` statically. 

Let me know if this looks good!

**Comment by Mmuzaf** [comment]
@arvindKandpal-ksolves , please also add a test to ensure that none of the commands which implement `LocalCommand` establish a connection. Otherwise looks good!

**Comment by arvindKandpal-ksolves** [comment]
Thanks for the feedback @Mmuzaf!

Added the test testLocalCommandsShouldNotConnect in NodetoolClassHierarchyTest 
to ensure all LocalCommand implementations do not establish a JMX connection.
Also restored the shouldConnect() override in History as it was needed for the test to pass.

Both HistoryTest and NodetoolClassHierarchyTest pass successfully.

**Review by smiklosovic — APPROVED** [review]

**Review by Mmuzaf — APPROVED** [review]

**Review by smiklosovic — APPROVED** [review]
